lib/kennel/output_limiter.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Kennel
+  class OutputLimiter
+    def initialize(out, max_lines, &on_overflow)
+      @out = out
+      @max_lines = max_lines
+      @on_overflow = on_overflow
```

Comment:
simpler when overflow is a string and then get `puts`-ed when overflow happens ?

Reply:
I'm not sure I see how it's simpler?

- it would lose the "&"
- s/call/puts/ (almost)

That's it, isn't it?

It would also be less flexible.